lib/stormpath/rails/config/social_login_verification.rb

```diff
@@ -0,0 +1,41 @@
+module Stormpath
+  module Rails
+    module Config
+      class SocialLoginVerification
+        attr_reader :app,
+                    :facebook_app_id,
+                    :facebook_app_secret,
+                    :github_app_id,
+                    :github_app_secret,
+                    :google_app_id,
+                    :linkedin_app_id
+
+        def initialize(app_href)
+          @app = Stormpath::Rails::Client.client.applications.get(app_href)
+          initialize_directories
+        end
+
+        private
+
+        def initialize_directories
```

Comment:
The ``AccountStoreMapping`` does have a search functionality, but unfortunately it is not directly associated with the ``provider``. It has access to the ``AccountStore`` (_has_one_) and then the ``AccountStore`` has access to the ``Provider``, so I can't do something like:

```ruby
app.account_store_mappings.search(account_store.provider.provider_id == 'facebook')
```

We can maybe refactor the _initialize_directories_ in this fashion?:

```ruby
app.account_store_mappings.map do |mapping|
  mapping.account_store if mapping.account_store.class == Stormpath::Resource::Directory
end.each do |directory|
  if directory.provider.provider_id == 'facebook'
    @facebook_app_id = directory.provider.client_id
    @facebook_app_secret = directory.provider.client_secret
  elsif directory.provider.provider_id == 'github'
    @github_app_id = directory.provider.client_id
    @github_app_secret = directory.provider.client_secret
  elsif directory.provider.provider_id == 'google'
    @google_app_id = directory.provider.client_id
  elsif directory.provider.provider_id == 'linkedin'
    @linkedin_app_id = directory.provider.client_id
  end
end
```